### PR TITLE
Check Vulkan 1.4 features for indexTypeUint8

### PR DESCRIPTION
**Pull request by lordalcol**

### CODE REVIEW

File: src/vulkan/device.cc [hunk 1]
```diff
@@ -767,6 +765,22 @@ Result Device::Initialize(
         return amber::Result("Missing compute full subgroups feature");
       }
     }
+
+    // If Vulkan 1.4 structure exists the features are set there.
```
Comment (dj2) [inline src/vulkan/device.cc:818]: What happens if vulkan 1.4 isn't provided, does this mean it can no longer find the feature it previously found? Does the old way not work anymore, like is `VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_INDEX_TYPE_UINT8_FEATURES_EXT` no longer returned above so the pointer isn't set?

File: src/vulkan/device.cc [hunk 2]
```diff
@@ -610,6 +614,10 @@ Result Device::Initialize(
       return amber::Result(
           "Subgroup extended types requested but feature not returned");
     }
+    if (feature == kAccelerationStructure &&
```
Comment (dj2) [inline src/vulkan/device.cc:617]: This seems unrelated?

Comment (lordalcol) [reply src/vulkan/device.cc:617]: yes, I think it's some merge gone bad. I'll re-do it

**Comment by lordalcol** [comment]
Abandoned in favour of https://github.com/google/amber/pull/1061
